statistics/histogram.go

```diff
@@ -508,7 +508,15 @@ func validRange(ran *ranger.Range) bool {
 	if ran.LowVal[0].Kind() == types.KindBytes {
 		low, high = ran.LowVal[0].GetBytes(), ran.HighVal[0].GetBytes()
 	} else {
-		low, high = codec.EncodeInt(nil, ran.LowVal[0].GetInt64()), codec.EncodeInt(nil, ran.HighVal[0].GetInt64())
+		var err error
+		low, err = codec.EncodeKey(nil, nil, ran.LowVal[0])
```

Comment:
In release 2.1, only int primary key will go to here. So pass a nil `stmtCtx` is okay.

Reply:
Seems not, in feedback.go:1106, it could be other types.

Reply:
No, `validRange` is only called by `SplitRange`.
And `SplitRange` is only call when converting table/index range to kv range or `dumpRangeFeedback` so other type won't be passed as a param.

Reply:
Yes, but in `dumpRangeFeedback` it could be other types.

Reply:
By passing a `StmtCtx` to `SplitRange()`, we can safely handle all the situations.

Reply:
PTAL @winoros